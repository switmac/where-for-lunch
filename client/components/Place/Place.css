@import 'styles/variables.css';

.name,
.link {
  margin-top: calc($base-spacing * 10);
  min-height: 2rem;
  font-size: 2rem;
  margin-bottom: calc($base-spacing * 2);
}
.link {
  color: white;
  text-decoration: none;
}
.box {
  min-height: 10rem;
  background-color: white;
  color: $black;
  background: linear-gradient(to bottom, rgba(255,255,255,1) 0%,rgba(255,255,255,0) 100%);
  border-radius: 1rem;
  padding: calc($base-spacing * 2);
  > div {
    margin-bottom: calc($base-spacing / 2);
  }
  .rating {
    margin-top: $base-spacing;
    position: realtive;
    display: inline;
  }
  .ratingScore, .stars {
    display: inline;
  }
  .stars {
    position: relative;
    vertical-align: middle;
    display: inline-block;
    color: $starGrey;
    overflow: hidden;
    margin-left: calc($base-spacing / 2);
  }
  .fullStars {
    position: absolute;
    left: 0;
    top: 0;
    white-space: nowrap;
    overflow: hidden;
    color: $starYellow;
  }
  .emptyStars:before, .fullStars:before {
    content:'\2605\2605\2605\2605\2605';
    font-size: 1rem;
  }
}

.photos {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  grid-auto-rows: 100px 100px;
  grid-auto-flow: dense;
  grid-gap: 10px;
}

.photo img {  
  object-fit: cover;
  height: 100%;
  width: 100%;
}
