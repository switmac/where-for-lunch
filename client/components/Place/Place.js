import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Place.css';

export default class Place extends PureComponent {
  static propTypes = {
    place: PropTypes.object,
    onClick: PropTypes.func,
  };

  handleOnClick = () => {
    const { place, onClick } = this.props;
    onClick && onClick(place);
  };

  render() {
    const { place, onClick } = this.props;
    const { name, hours, photos, address, phone, categories, price, rating } = place;
    const header = place.name ? <Link to="/place" className={styles.link} onClick={ this.handleOnClick }>{ name }</Link> : 'Where for lunch?';
    const photoList = photos ? photos.map((photo, index) => {
      return <div key={ index } className={styles.photo}><img src={ photo } /></div>;
    }) : '';
    const businessHours = hours ? hours.map((hour, index) => {
      return <div key={ index } className={ styles.storeHours }>
        <div className={ styles.hoursType }> { hour.hoursType }</div>
        <div className={ styles.isOpenNow }> { hour.isOpenNow ? 'OPEN NOW' : 'CLOSE NOW'}</div>
      </div>;
    }) : '';


    return (
      <div className={styles.root} >
        <div className={styles.name}>{ header } </div>
        <div className={styles.box}>
          <div>{ address }</div>
          <div>{ phone }</div>
          <div>{ categories && categories.join(', ') }</div>
          <div>{ price }</div>
          { rating &&
            <div className={styles.rating}>
              <div className={styles.ratingScore}>{ place.reviewCount } reviews</div>
              <div className={styles.stars}>
                <div className={styles.emptyStars}></div>
                <div className={styles.fullStars} style={{ width: `${place.rating / 5 * 100}%` }}></div>
              </div>
            </div>
          }
          { businessHours && !onClick &&
            <div className={styles.businessHours}>
              { businessHours }
            </div>
          }
          { photoList && !onClick &&
            <div className={styles.photos}>
              { photoList }
            </div>
          }
        </div>
      </div>
    );
  }
}
