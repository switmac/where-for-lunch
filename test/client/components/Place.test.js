import React from 'react';
import { shallow } from 'enzyme';
import Place from 'client/components/Place/Place';
import renderer from 'react-test-renderer';

test('With Enzyme, Place component not render rating section when no rating passed over', () => {
  const place = { hehe: 'haha' };
  const wrapper = shallow(
    <Place place={place} />,
  );
  const p = wrapper.find('.rating');
  expect(p.length).toBe(0);
});

test('With Jest snapshot, Place component renders rating section when present', () => {
  const place = { hehe: 'haha', rating: 3.5, reviewCount: 10 };
  const placeComponent = renderer
    .create(<Place place={place} />)
    .toJSON();
  expect(placeComponent).toMatchSnapshot();
});
