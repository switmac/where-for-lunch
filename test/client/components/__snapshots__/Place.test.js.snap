// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Place component renders rating section when present 1`] = `
<div
  className="root"
>
  <div
    className="name"
  >
    Where for lunch?
  </div>
  <div
    className="box"
  >
    <div />
    <div />
    <div />
    <div />
    <div
      className="rating"
    >
      <div
        className="ratingScore"
      >
         reviews
      </div>
      <div
        className="stars"
      >
        <div
          className="emptyStars"
        />
        <div
          className="fullStars"
          style={
            Object {
              "width": "70%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
